src/draw/lv_draw_vector.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file lv_draw_vector.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_draw_vector.h"
+#include "../stdlib/lv_mem.h"
+#include "../stdlib/lv_string.h"
+#include <math.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#ifndef M_PI
+    #define M_PI 3.1415926f
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC FUNCTIONS
+ **********************/
+
+static inline void _matrix_translate(float m[3][3], float x, float y)
+{
+    m[0][2] += m[0][0] * x + m[0][1] * y;
+    m[1][2] += m[1][0] * x + m[1][1] * y;
+    m[2][2] += m[2][0] * x + m[2][1] * y;
+}
+
+static inline void _matrix_scale(float m[3][3], float a, float b)
+{
+    m[0][0] *= a;
+    m[1][0] *= a;
+    m[2][0] *= a;
+    m[0][1] *= b;
+    m[1][1] *= b;
+    m[2][1] *= b;
+}
+
+static inline void _matrix_rotate(float m[3][3], float c, float s)
+{
+    m[0][0] = c * m[0][0] + s * m[0][1];
+    m[1][0] = c * m[1][0] + s * m[1][1];
+    m[2][0] = c * m[2][0] + s * m[2][1];
+    m[0][1] = c * m[0][1] - s * m[0][0];
+    m[1][1] = c * m[1][1] - s * m[1][0];
+    m[2][1] = c * m[2][1] - s * m[2][0];
+}
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_draw_vector_dsc_init(lv_draw_vector_dsc_t * dsc)
+{
+    LV_PROFILER_BEGIN;
+    lv_memzero(dsc, sizeof(lv_draw_vector_dsc_t));
+    dsc->fill_dsc.opa = LV_OPA_COVER;
+    dsc->fill_dsc.img_dsc.zoom = LV_ZOOM_NONE;
+#if LV_COLOR_DEPTH > 8
+    dsc->fill_dsc.img_dsc.antialias = 1;
+#endif
+    dsc->fill_dsc.matrix.m[0][0] = 1.0f;
+    dsc->fill_dsc.matrix.m[1][1] = 1.0f;
+    dsc->fill_dsc.matrix.m[2][2] = 1.0f;
+    dsc->matrix.m[0][0] = 1.0f;
+    dsc->matrix.m[1][1] = 1.0f;
+    dsc->matrix.m[2][2] = 1.0f;
+    dsc->stroke_dsc.miter_limit = 4;
+    dsc->scissor_area.x2 = LV_COORD_MAX;
+    dsc->scissor_area.y2 = LV_COORD_MAX;
+    LV_PROFILER_END;
+}
+
+void lv_draw_vector(struct _lv_layer_t * layer, lv_vector_path_t * path, lv_draw_vector_dsc_t * dsc)
```

Comment:
```suggestion
void lv_draw_vector(struct _lv_layer_t * layer, lv_draw_vector_dsc_t * dsc, lv_vector_path_t * path)
```

`lvgl` follow the pattern of draw(layer, dsc, other_args).
See:
```c

/**
 * Draw an image
 * @param draw_ctx      pointer to the current draw context
 * @param dsc           pointer to an initialized `lv_draw_img_dsc_t` variable
 * @param coords        the coordinates of the image
 * @param src           pointer to a lv_color_t array which contains the pixels of the image
 */
void lv_draw_img(struct _lv_layer_t * layer, const lv_draw_img_dsc_t * dsc, const lv_area_t * coords);


void lv_draw_layer(struct _lv_layer_t * layer, const lv_draw_img_dsc_t * dsc, const lv_area_t * coords);
LV_ATTRIBUTE_FAST_MEM void lv_draw_label(lv_layer_t * layer, const lv_draw_label_dsc_t * dsc,
                                         const lv_area_t * coords);

LV_ATTRIBUTE_FAST_MEM void lv_draw_letter(lv_layer_t * layer, lv_draw_label_dsc_t * dsc,
                                          const lv_point_t * point, uint32_t unicode_letter);

```

Reply:
done